analytics-ios-mcvid/Classes/SEGMCVIDTracker.m

```diff
@@ -32,10 +37,169 @@ -(id)initWithOrganizationId:(NSString *)organizationId region:(NSString *)region
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+    NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:config];
+
+    [[session dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        void (^callbackWithCode)(MCVIDAdobeErrorCode code, NSString *message, NSError *error) = ^void(MCVIDAdobeErrorCode code, NSString *message, NSError *error) {
+            MCVIDAdobeError *adobeError = [[MCVIDAdobeError alloc] initWithCode:code message:message error:error];
+            NSError *compositeError = [NSError errorWithDomain:errorDomain code:adobeError.code userInfo:@{MCVIDAdobeErrorKey:adobeError}];
+            completion(compositeError);
+        };
+
+
+        NSDictionary *dictionary = nil;
+        @try {
+            dictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+        } @catch (NSException *exception) {
+            return callbackWithCode(MCVIDAdobeErrorCodeClientSerializationError, @"Deserializing the JSON response failed", nil);
+        }
+        
+        // or { ..., "errors": [{ "code": 2, "msg": "error" } ... ], ... }
+        NSError *errorObject = dictionary[errorResponseKey][0];
+        NSString *errorMessage = dictionary[@"errors"][0][@"msg"];
+        
+
+        NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
+        NSInteger responseStatusCode = [httpResponse statusCode];
+        NSUInteger milliSecondsToWait = currentRetryCount * currentRetryCount;
+        NSUInteger milliSecondsWaited = currentRetryCount * (currentRetryCount + 1)  * ((2 * currentRetryCount) + 1)/6;
+        
+        if ((currentRetryCount > maxRetryCount) && (errorObject)) {
+            return callbackWithCode(MCVIDAdobeErrorCodeServerError, errorMessage, errorObject);
+
+        }
+        
+        if (milliSecondsWaited / 1 >= maxRetryTimeSecs && (errorObject)) {
+            return callbackWithCode(MCVIDAdobeErrorCodeServerError, errorMessage, errorObject);
+        }
+        
+        if ((responseStatusCode == 200) && (!errorObject)){
+            completion(nil);
+        } else {
+            dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_SEC * milliSecondsToWait);
+            dispatch_after(delay, self.backgroundQueue, ^(void){
+                [self syncMarketingCloudId:advertisingId organizationId:organizationId marketingCloudId:marketingCloudId maxRetryCount:maxRetryCount currentRetryCount:currentRetryCount+1 maxRetryTimeSecs:maxRetryTimeSecs completion:^(NSError *error){
+                }];
+            });
+        }
+    }] resume];
+}
+
+- (NSURL *)createURL:advertisingId organizationId:(NSString *)organizationId marketingCloudId:(NSString *)marketingCloudId {
```

Comment:
💡 Taking into account the changes we need in order to make our own id-sync calls, you will probably want to change this method into something like:

``` suggestion
- (NSURL *)URLFromOrganizationId:(NSString *)organizationId marketingCloudId:(NSString *)marketingCloudId IntegrationCode:(NSString *)integrationCode userIdentifier:(NSString *)userIdentifier {
```

The last two parameters would be optional and be used only by your idsync call, passing `DSID_20915` as the `integrationCode` and the advertising identifier as the `userIdentifier`. Inside this method you would then concatenate them using the `%01` separator just like you do with your existing `deviceTypeKey` and `advertisingId`.